src/main/java/io/appium/java_client/pagefactory/AppiumFieldDecorator.java

```diff
@@ -87,12 +87,12 @@ public AppiumFieldDecorator(SearchContext context, long timeout,
      * @param duration is a desired duration of the waiting for an element presence.
      */
     public AppiumFieldDecorator(SearchContext context, TimeOutDuration duration) {
-        this.originalDriver = unpackWebDriverFromSearchContext(context);
-        HasSessionDetails hasSessionDetails = ofNullable(this.originalDriver).map(webDriver -> {
-            if (!HasSessionDetails.class.isAssignableFrom(originalDriver.getClass())) {
+        this.webDriver = unpackWebDriverFromSearchContext(context);
+        HasSessionDetails hasSessionDetails = ofNullable(this.webDriver).map(webDriver -> {
```

Comment:
I meant it's better to use the mapped local variable `webDriver ->` inside the lambda handler rather than class-level `this.originalDriver`one.